### ISSUE TITLE
Invalid adtag

### ISSUE THREAD

**Issue by aa1008611** [geovex/tgp#3]
adtag

**Comment by geovex** [comment]
This mode is experimental. Can you provide full config?

**Comment by aa1008611** [comment]
listen_url = "0.0.0.0:666"
obfuscate = true
host = "cloudflare.com:443"
ignore_timestamp = true
secret = "ee00000000000000000000000000000000"
adtag = "ba87f7ccbb0e56c530e8d8a50e4b6b95"

**Comment by geovex** [comment]
As unpleasant as it may sound, it "works for me". Your config seems correct (except obfuscate option is useless with adtag). I can only advise you to check all settings in `MTProxybot` (may be there are typos in IP or channel name) and also check stats so you can see whether clients' connections are active at all. 

Since this mode has mediocre implementation under the hood it will be experimental for a long time. You can use [official proxy server](https://github.com/TelegramMessenger/MTProxy) which is specifically designed for such scenario or [this alternative client](https://github.com/alexbers/mtprotoproxy).

**Comment by aa1008611** [comment]
![Screenshot_20231025_103048](https://github.com/geovex/tgp/assets/88304239/6c9a674c-a2ef-4bb2-94fa-81401f00bcaf)
![Screenshot_20231025_103058](https://github.com/geovex/tgp/assets/88304239/0427fba1-3c07-405d-b31b-0d14a794b3e1)
![Screenshot_20231025_103102](https://github.com/geovex/tgp/assets/88304239/22460778-43f0-47ce-ac93-909a77a21933)
![Screenshot_20231025_103127](https://github.com/geovex/tgp/assets/88304239/a2c043c4-f1ed-43b0-8085-88ae928f259a)


No, there are no errors. This is how I configured it, and other agents can display channels. There are no errors. You can take a look and configure it yourself. Try to see if the channel can be displayed,

**Comment by aa1008611** [comment]
![IMG_20231025_214233](https://github.com/geovex/tgp/assets/88304239/03166a17-b9f0-43a2-b019-fc9d7da5a16e)

Proxy x.x.x.x:5555 with the tag 5b114c49932a72515d800ad1e2d7d9ee.
Promoted channel: xxxx .
How can I help?

I switched to another one and it still doesn't show., The above is the generated tag without any errors,. You can test it out

**Comment by geovex** [comment]
Actually I tried your tag and desktop client shows sponsored channel. My key hypothesis is that this is some wired region-related issue.

**Comment by aa1008611** [comment]
I can connect to the agent, but do not display sponsors, so is this my problem or the project's problem? Other projects can display label sponsors

**Comment by geovex** [comment]
My last ridiculous assumption is that you already own/subscribed-to channel and test it from the same account, and that's why you don't see differences. Try check from other account of from friend's phone.

If I assume that everything you can, you did right, then it either some weird regional ban (unlikely for channel with 1-2k users) or some corner case that I can not catch (again, i see your advertised channel when I use your adtag in my config, this means you registered it correctly and it should work).

So far we established:
You registered proxy and sponsored channel.
Your sponsored channel works on all my test devices (means you did that part right).
At least one of your devices does not advertise or you can't see it for some reason. 

Suggested solution: try check from different account and different device. 

<details> 
  <summary>some technical details</summary>

   Proxy does not know about sponsorship. It can push adtag upstream during connection and then it just forwards encrypted messages. The other notable fact is that with adtag tgp operates on very different algorithm because it has to connect to different upstream servers and distinguish separate messages. In layman's terms if connection established, everything should work and tgp only handles the connection.

  Main goals of this particular software was to add multi-user support and ability to connect to upstream through proxy or anonymizer. Sponsored mode (requires connection through middle proxies) was added as a bonus in spare time and i rarely check this functionality.

If this functionality is critical for you, you can try other software implementations.

</details>

**Comment by aa1008611** [comment]
I did a great job testing multiple accounts, but they didn't show up.. I don't know what happens when I connect to your agent

**Comment by geovex** [comment]
I'm afraid a can not determine cause of the issue. I will left this open until something new comes up.

**Comment by aa1008611** [comment]
Thank you for your answer. wish you good luck

**Comment by DzmitryKudrashou** [comment]
I've just checked it. It works well for me. Is it possible that it depends on some variables that we don't know about?

**Comment by aa1008611** [comment]
> I've just checked it. It works well for me. Is it possible that it depends on some variables that we don't know about?

Location of use in China
